src/test/java/de/retest/ui/descriptors/DefaultRetestIdProviderTest.java

```diff
@@ -0,0 +1,88 @@
+package de.retest.ui.descriptors;
+
+import static de.retest.ui.Path.fromString;
+import static de.retest.ui.descriptors.IdentifyingAttributes.create;
+import static de.retest.ui.descriptors.IdentifyingAttributes.createList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collection;
+
+import org.junit.Test;
+
+public class DefaultRetestIdProviderTest {
+
+	@Test
+	public void too_long_text_should_be_cut() {
+		final DefaultRetestIdProvider cut = new DefaultRetestIdProvider();
+
+		// for what you would actually call a text
+		final IdentifyingAttributes ident = createIdentAttributes(
+				"This is some very long sentence, that could be in a link text, or in some paragraph, and really is to long to be used as id." );
+		final String retestId = cut.getRetestId( ident );
+		assertThat( retestId.length() ).isLessThan( 20 );
+
+		// and cut should still be unique!
+		assertThat( retestId.length() ).isNotEqualTo( cut.getRetestId( ident ) );
+	}
+
+	@Test
+	public void too_long_words_should_be_cut() {
+		final DefaultRetestIdProvider cut = new DefaultRetestIdProvider();
+
+		// but also for single words
+		final IdentifyingAttributes ident = createIdentAttributes( "supercalifragilisticexpialidocious" );
+		final String retestId = cut.getRetestId( ident );
+		assertThat( retestId.length() ).isLessThan( 20 );
+
+		// and cut should still be unique!
+		assertThat( retestId.length() ).isNotEqualTo( cut.getRetestId( ident ) );
+	}
+
+	public IdentifyingAttributes createIdentAttributes( final String text ) {
+		final Collection<Attribute> attributes = createList( fromString( "/HTML/DIV" ), "DIV" );
+		attributes.add( new StringAttribute( "text", text ) );
+		final IdentifyingAttributes ident = new IdentifyingAttributes( attributes );
+		return ident;
+	}
+
+	@Test
+	public void should_always_be_unique() {
+		final DefaultRetestIdProvider cut = new DefaultRetestIdProvider();
+		final IdentifyingAttributes ident = createIdentAttributes( "a" );
+		final String retestId = cut.getRetestId( ident );
+		assertThat( retestId ).isNotEqualTo( cut.getRetestId( ident ) );
+		assertThat( retestId ).isNotEqualTo( cut.getRetestId( ident ) );
+		assertThat( retestId ).isNotEqualTo( cut.getRetestId( ident ) );
+		assertThat( retestId ).isNotEqualTo( cut.getRetestId( ident ) );
```

Comment:
Same check 4 times?

Reply:
getRetestId should return a new unique ID every time...